rero/rero-ils-ui
# autocomplete: fix suggestions list

Old version of the third party library `ngx-bootstrap` had sometime weird
behaviors. `@rero/ng-core@0.2.3` fix this.

* Fixes `ngx-bootstrap` imports to be angular 9 and `ngx-bootstrap>6`
compliant.
* Fixes tests form other pull requests by mocking the services.
* Updates dependency @rero/ng-core to version 0.2.3.
* Fixes the number of suggestions for each category documents and direct
links to be sure that all suggestions are displayed.

Note: several versions of `ngx-bootstrap` has been tried, but some of
them failed:
- 3.3.x: does not display a `typeahead` suggestion list for slow query.
- 3.6.1: does not compile our project in production.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Dependencies

- ng-core: https://github.com/rero/ng-core/pull/177

## Why are you opening this PR?

- Fixes: https://github.com/rero/rero-ils/issues/964

## How to test?

- Try all ui widget component in the professional and public view. Especially the typeahead.
- Editors should not be concerned.

## Code review check list
